src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManagerTest.java

```diff
@@ -92,4 +96,81 @@ public void getNextQuestionsForFilter_appliesExclusionFilterForDeprecatedQuestio
     assertEquals(deprecatedFilter.getOperator(), Constants.BooleanOperator.NOT);
     assertEquals(deprecatedFilter.getValues(), Collections.singletonList("true"));
   }
+
+  @Test
+  public void generateRandomQuestions_returnsCorrectNumberOfQuestions() throws ContentManagerException {
+
+    // Arrange
+    int limit = 5;
+
+    var gameManager = new GameManager(
+        this.dummyContentManager,
+        this.dummyGameboardPersistenceManager,
+        this.dummyMapper,
+        this.dummyQuestionManager,
+        "latest"
+    );
+
+    List<ContentDTO> questions = new ArrayList<>();
+    for (int i = 0; i < limit; i++) {
+      IsaacQuestionPageDTO question = createMock(IsaacQuestionPageDTO.class);
+      // Setting the expected behavior for the getSupersededBy method
+      expect(question.getSupersededBy()).andStubReturn(null);
+      questions.add(question);
+    }
+    var resultsWrapper = new ResultsWrapper<>(questions, (long) limit);
+
+    expect(dummyContentManager.findByFieldNamesRandomOrder(
+        anyObject(),
+        anyInt(),
+        anyInt(),
+        anyLong())
+    ).andReturn(resultsWrapper)
+        .times(1);
+    replay(dummyContentManager);
+
+    // Act
+    var result = gameManager.generateRandomQuestions(new GameFilter(), limit);
+
+    // Assert
+    // Check that the result has the correct number of questions
```

Comment:
As it stands, this test is not very useful, because of course the size of the result array is going to be the same as `limit`: you've configured the dummy content manager to return that exact number of questions. A more useful test would be to configure the dummy content manager to return 20 questions, for instance. And then you check that the result size is `limit` (so the same assertion you're doing right now). That way, you can see how the list of questions returned by the content manager are cropped so that they're never greater than the specified limit.